circleci/templates/maven/config_template-v1.yml

```diff
@@ -49,6 +49,15 @@ jobs:
           name: Maven test phase
           command: |
             mvn test -B $DISABLE_DOWNLOAD_PROGRESS_OPTS
+      - run:
```

Comment:
It be obvious, but can you place some comments in here as to why we do this and effectively what it does?

To expound:
- I see it makes a directory in the home location called test-results/junit/ (don't remember what -p does)
- Then it finds anything in the current directory meaning the regex looking for specifically a test-results folder from the build and xml file to then cp to the previous made directory.  
-Seems like we tell it to do this always, do we have to tell it this?  What happens if we don't?
- Then you have it store both the test-results dir & the junit inside the test results
--> How long does that storage keep around for?  (the workflow, the job?)
--> Why are we storing both the parent and child directories?  Does the store commands not do it naturally?  Do they do something slightly different?

Of course, there is a good chance that reading the CircleCI doc I can figure this out possibly.  But I would like to spell it out a bit and talk about where this is consumed as well.

Reply:
The doc helps.

 -p is used to create parent dirs where they don't exist instead of erroring

JUnit specifically puts XML results in a defined place, but CircleCI caters to many test suites and wants them to be moved into a cental location. If we don't tell it to move anything, CircleCI won't scan any additional directories and results won't be uploaded.

Store test results in test-results and artifacts in test-results/junit, because there could be multiple types of tests.